src/VisualStudio/Core/Impl/Options/GridOptionPreviewControl.xaml.cs

```diff
@@ -93,5 +97,25 @@ internal override void Close()
                 this.ViewModel.Dispose();
             }
         }
+
+        private void Generate_Editorconfig(object sender, System.Windows.RoutedEventArgs e)
+        {
+            var optionSet = this.ViewModel.ApplyChangedOptions(this.OptionService.GetOptions());
+            var editorconfig = _getCurrentEditorConfigOptionsString(optionSet);
+
+            var sfd = new System.Windows.Forms.SaveFileDialog
+            {
+                Filter = "All files (*.*)|",
+                FileName = ".editorconfig",
+                Title = "Save .editorconfig File"
+            };
+            if (sfd.ShowDialog() == System.Windows.Forms.DialogResult.OK)
+            {
+                var path = sfd.FileName;
+                var sw = new StreamWriter(File.Create(path));
+                sw.Write(editorconfig);
+                sw.Close();
```

Comment:
1. all this needs to be wrapped in IOUtilities.PerformIO
2. "using (var streamWriter = ...)"

What does File.Create do if the path already exists?